prow/cmd/exporter/collector.go

```diff
@@ -60,7 +60,17 @@ func (pjc prowJobCollector) Collect(ch chan<- prometheus.Metric) {
 	}
 	//We need to filter out the latest jobs
 	//because sending the same sample twice would lead to prometheus runtime error
-	for _, pj := range getLatest(prowJobs) {
+	latestJobs := getLatest(prowJobs)
+
+	//sort by keys for stable ordering required by unit test
+	keys := make([]string, 0, len(latestJobs))
+	for k := range latestJobs {
+		keys = append(keys, k)
+	}
+	sort.Strings(keys)
```

Comment:
Couldn't we sort after we filter? Then we sort a set that's at most as large as the total amount of possible jobs, whereas this will sort something that is as large as every run of every job in the cluster?

Reply:
`getLatest(prowJobs)` is the filter, right?

Reply:
Fix the test itself since we know the core code is working.